src/Chapter3.hs

```diff
@@ -480,6 +516,37 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data ChurchOrLibrary = Church | Library
+
+data Castle = MkCastle
+  { castleBuilt :: Bool
+  , castleName :: String 
+  } deriving (Show)
+
+data City = MkCity
+  { cityCastle :: Castle
+  , cityWall :: Bool
+  , cityChurch :: ChurchOrLibrary
+  , cityHouses :: [Int]
+  }
+
+getPopulation :: [Int] -> Int
+getPopulation (x:xs) = if length xs == 0 then x else x + getPopulation xs
```

Comment:
This function is partial because it doesn't pattern match on the empty list. You can make your function handle all cases and also more efficient, if you add a special case like:

```haskell
getPopulation [] = 0
```

After that change, your recursive implementation will become much simpler :slightly_smiling_face: Morever, this is a standard function called `sum`